### PR TITLE
[FIF-213] Add TeamCity build project for ETL

### PR DESCRIPTION
**Pull request by kurtjohnson-HS**
This PR adds a Team City build for the ETL project with a NuGet archive artifact.

### CODE REVIEW

File: .teamcity/ETL/Project.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    buildType(etl.buildTypes.PullRequestETLBuild)
+    buildType(etl.buildTypes.BranchETLBuild)
+    buildType(etl.buildTypes.DeployETLBuild)
+
+    params{
+        param("project.directory", "./EdFi.Buzz.Etl");
+        param("octopus.release.version","<placeholder value>")
+        param("octopus.release.project", "Buzz ETL")
+        param("octopus.project.id", "Projects-111")
```
Comment (stephenfuqua) [inline .teamcity/ETL/Project.kt:23]: This project id may need to change at some point. One of us needs to get around to doing more work in Octopus Deploy. I told David I would do that a few weeks ago and I haven't been able to prioritize it yet. Still hoping to do so soon.

So no action to take here. Just calling it out as a reminder.

File: .teamcity/ETL/buildTypes/DeployETL.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.swabra
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildSteps.nuGetPublish
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildSteps.powerShell
+
+
+object DeployETLBuild : BuildType ({
+    name = "Deploy"
+
+    features {
+        // Default setting is to clean before next build
```
Comment (stephenfuqua) [inline .teamcity/ETL/buildTypes/DeployETL.kt:18]: FYI, recently I realized that this default setting is a "swabra clean", not a "git clean". There is subtle difference between the two. The default setting is generally good enough, but occasionally will need to be modified to force a `git clean`. 

No action to take here.

File: EdFi.Buzz.Etl/eng/Windows/install.ps1 [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+  [string]
+  [Parameter(Mandatory = $true)]
+  $DbPassword,
+
+  [string]
+  $DbName = "edfi_buzz"
+
+)
+
+# TODO: insure this is run as an administrator
```
Comment (stephenfuqua) [inline EdFi.Buzz.Etl/eng/Windows/install.ps1:35]: Please remove - line 7 does the trick. I missed that update in one of the other files.

File: EdFi.Buzz.Etl/eng/Windows/install.ps1 [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+# TODO: ensure node.js exists with minimum version
+
+# TODO: refactor to share functions instead of duplicate them
+# in various application installer scripts.
```
Comment (stephenfuqua) [inline EdFi.Buzz.Etl/eng/Windows/install.ps1:40]: Can you create two new FIF tickets (low priority) for each of these TODO statements, and then remove the TODO statements?

File: EdFi.Buzz.Etl/eng/build.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+// See the LICENSE and NOTICES files in the project root for more information.
+
+// This build script runs a number of yarn commands in an OS-agnostic fashion
+// with the help of Node.js.
+
+const { execSync } = require('child_process');
+
+const steps = [
+  'yarn synp -s yarn.lock -f',
+  'yarn copyfiles .env.example package-lock.json package.json config/**/* processors/**/*  src/**/* sql/**/* tasks/**/* dist',
```
Comment (stephenfuqua) [inline EdFi.Buzz.Etl/eng/build.js:13]: This is a larger set of files than I would have expected. The original use of this was to copy a few files that `yarn build` did not load into `dist`, but which are required on the deployment server. I'm particularly curious about `src/**/*` - why would source TypeScript files need to be put into the distribution?

File: EdFi.Buzz.Etl/eng/edfi.buzz.etl.nuspec [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  </metadata>
+  <files>
+    <file src=".\Windows\*" target="windows" />
+    <!-- NB: nuget.exe does not handle widlcards with forward-slash paths. Must use back-slash. -->
+    <file src="..\dist\*.json" target="dist" />
+    <file src="..\dist\config\**" target="dist/config" />
+    <file src="..\dist\processors\**" target="dist/processors" />
+    <file src="..\dist\sql\**" target="dist/src" />
+    <file src="..\dist\src\**" target="dist/src" />
+    <file src="..\dist\tasks\**" target="dist/tasks" />
```
Comment (stephenfuqua) [inline EdFi.Buzz.Etl/eng/edfi.buzz.etl.nuspec:22]: Do these need to be called out separately, or will it work with just 

```
<file src="..\dist" target="dist/" />
```

Now that I'm thinking about it, I think that ended up either not being recursive or it dumped everything into the same folder. Can you confirm?

File: EdFi.Buzz.Etl/eng/build.js (hunk 5 above)
Comment (kurtjohnson-HS) [reply EdFi.Buzz.Etl/eng/build.js:13]: I cannot speak for Xavier, but resources for completing the database ETL in TypeScript were having an impact on the delivery time. I opted for straight JavaScript to ensure I could complete it on time.

File: EdFi.Buzz.Etl/eng/edfi.buzz.etl.nuspec (hunk 6 above)
Comment (kurtjohnson-HS) [reply EdFi.Buzz.Etl/eng/edfi.buzz.etl.nuspec:22]: We can use the directory file listing. It does create the dist folder with the files in their proper folders.

File: EdFi.Buzz.Etl/eng/build.js (hunk 5 above)
Comment (stephenfuqua) [reply EdFi.Buzz.Etl/eng/build.js:13]: OK

**Pull request merged**